common.c

```diff
@@ -30,6 +32,31 @@
 static void log_error(const char *fmt, ...)
 	__attribute__ ((format (printf, 1, 2)));
 
+
+static const char* get_curtime_string()
+{
+	// This will invalidate previous return pointer on the next call.
+	// Doesn't really matter for us, since we're using it immediately.
+	static char buffer[40];
+
+	// https://stackoverflow.com/a/32983646
+	struct timeval tv;
+	gettimeofday(&tv, NULL);
+
+	int millisec = tv.tv_usec/1000;
+	if (millisec>=1000) { // Allow for rounding up to nearest second
+		millisec -=1000;
+		tv.tv_sec++;
+	}
```

Comment:
This doesn't seem to look right. I don't think the `if` branch will ever be hit - `millisec` could be `1000` only if it was rounded up to that value following a floating point division, which is what the SO answer does but is not done here.

Honestly I would suggest to not bother with rounding up milliseconds and just do an integer division by `1000`.